vendor/engines/secure_rooms/app/controllers/secure_rooms_api/scans_controller.rb

```diff
@@ -8,18 +8,21 @@ class SecureRoomsApi::ScansController < ApplicationController
   before_action :load_user_and_reader
 
   def scan
-    accounts = @user.accounts_for_product(@card_reader.secure_room)
-    requested_id = params[:account_identifier].to_i
-    selected_account = accounts.find { |account| account.id == requested_id }
+    requested_account_id = params[:account_identifier].to_i
 
-    access_verdict = SecureRooms::CheckAccess.new.authorize(
+    verdict = SecureRooms::CheckAccess.new.authorize(
       @user,
       @card_reader,
-      accounts: accounts,
-      selected_account: selected_account,
+      requested_account_id: requested_account_id,
     )
 
-    render SecureRooms::ScanResponsePresenter.new(@user, @card_reader, access_verdict, accounts).response
+    SecureRooms::AccessManager.new(
+      verdict,
+    ).process
+
+    render SecureRooms::ScanResponsePresenter.new(
+      verdict,
+    ).response
```

Comment:
On that same note, `SecureRooms::ScanResponsePresenter.present(verdict)` could be a nice api.